src/base/PolymeshTransactionBase.ts

```diff
@@ -370,7 +371,7 @@ export abstract class PolymeshTransactionBase<
     await context.assertHasSigningAddress(signingAddress);
 
     // era is how many blocks the transaction remains valid for, `undefined` for default
-    const era = mortality.immortal ? 0 : mortality.lifetime?.toNumber();
+    const era = mortality.immortal ? 0 : (mortality.lifetime?.toNumber() as number);
```

Comment:
The `?.` can be `undefined` so casting to a number looks wrong to me. This is working around another rule of that prevents the normal `!` cast that was used.